src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -614,6 +614,9 @@ bool Creature::UpdateEntry(uint32 entry, CreatureData const* data /*= nullptr*/,
         ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_ATTACK_ME, true);
     }
 
+    if (cInfo->flags_extra & CREATURE_FLAG_EXTRA_NO_COMBAT)
```

Comment:
What if it'd been already modified to `_isIgnoringCombat = true`? Unnecessary call of `SetIgnoringCombat(true)` function.

Reply:
Wrong. UpdateEntry is being called on creature creation and transforms which both are suposed to consider this flag since transform entries can have different flags than the original entry.